### PR TITLE
storage: limit key size.

### LINKED ISSUE #2438: add limitation for key size
We limit value size to 5 MB, and we should also limit key size to a small size. 

Supporting too large key is not a good idea, this may cause huge heartbeat message, and may also reduce the write performance for RocksDB.

TiDB limits key size to 3 KB, we can limit it to 4 KB

### PR DESCRIPTION
**Pull request by disksing**
Fix #2438

### CODE REVIEW

File: src/storage/mod.rs [hunk 1]
```diff
@@ -653,6 +655,13 @@ impl Storage {
         options: Options,
         callback: Callback<Vec<Result<()>>>,
     ) -> Result<()> {
+        for m in &mutations {
+            let (size, limit) = (m.key().encoded().len(), self.max_key_size);
```
Comment (siddontang) [inline src/storage/mod.rs]: seem needless to define the limit variable.

**Comment by siddontang** [comment]
PTAL @zhangjinpeng1987

File: src/storage/mod.rs (hunk 1 above)
Comment (disksing) [reply src/storage/mod.rs]: `Error::KeyTooLarge { size, self.max_key_size }` will not compile, rustc complains something about 'expect self.max_key_size to be an identifier' :(

**Comment by siddontang** [comment]
LGTM

**Pull request merged**